laravel/app/Models/Schedule.php

```diff
@@ -147,15 +147,23 @@ static public function setTimes($input)
         return $input;
     }
 
-    // Helper function to format timestamps before displaying them in forms
+    // Helper function to format timestamps before submitting them to the server
     public function formatTimes()
     {
         $start = date_parse_from_format('H:i', $this->start_time);
         $end = date_parse_from_format('H:i', $this->end_time);
         $duration = date_parse_from_format('H:i', $this->duration);
 
+        //check for any default start hours and if there are, don't pad start hours
+        (!in_array($start['hour'], [0,6,9,12,24])) ?
+        $this->start_time = str_pad($start['hour'], 2, 0, STR_PAD_LEFT) . ":" . str_pad($start['minute'], 2, 0, STR_PAD_LEFT) :
         $this->start_time = $start['hour'] . ":" . str_pad($start['minute'], 2, 0, STR_PAD_LEFT);
-        $this->end_time = $end['hour'] . ":" . str_pad($end['minute'], 2, 0, STR_PAD_LEFT);
+        
+        //check for any default duration hours and if there are, don't pad duration hours
```

Comment:
Why would you do this? Can't you just fix the front-end to check for padded zeroes?

Reply:
Simply edit the template file - https://github.com/playasoft/volunteers/blob/master/laravel/resources/views/pages/schedule/edit.blade.php#L120-L121